plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLspService.kt

```diff
@@ -101,6 +107,10 @@ internal class LSPProcessListener : ProcessListener {
 
 @Service(Service.Level.PROJECT)
 class AmazonQLspService(private val project: Project, private val cs: CoroutineScope) : Disposable {
+    private val serverStartedListener = mutableListOf<AmazonQServerStartedListener>()
+    fun addServerStartedListener(listener: AmazonQServerStartedListener) = serverStartedListener.add(listener)
+    fun notifyServerStarted() = serverStartedListener.forEach { it() }
```

Comment:
would prefer to have a flag that can be queried by consumers to handle the case where the server exits unexpectedly

Reply:
This has been renamed incorrectly. This listens to when we receive an initialize message and is separate from starting servers. Monitoring server state will be executed separately